plugins/python/io/NumpyReader.cpp

```diff
@@ -147,6 +166,34 @@ PyArrayObject* load_npy(std::string const& filename)
     return reinterpret_cast<PyArrayObject*>(array);
 }
 
+PyArrayObject* load_npy_script(std::string const& source,
+                               std::string const& module,
+                               std::string const& function,
+                               std::string const& fargs)
+{
+
+    MetadataNode m;
+    std::unique_ptr<plang::Script> script(new plang::Script(source, module, function));
+    std::unique_ptr<plang::Invocation> method(new plang::Invocation(*script, m, fargs));
```

Comment:
I don't understand this.  Can't you just call the method with the args without using Script and Invocation?  It seems like that's all you're doing in the end anyway.

Reply:
Invocation is what does the compilation of the user's script. We need the code they gave us (through `m_filename` or as the `source` option, compiled to Python objects before we can call any functions.